src/prefect/engine.py

```diff
@@ -1340,6 +1343,7 @@ async def begin_task_run(
                 # When a a task run finishes on a remote worker flush logs to prevent
                 # loss if the process exits
                 APILogHandler.flush(block=True)
+                await client.disconnect_events()
```

Comment:
This lines up with my (limited) understanding of how we make sure to flush all our run logs to the server, and is where I was expecting we'd tie into the current lifecycle of tasks and flows.

Reply:
👍 these make sense to me